functional_tests/functional_test.go

```diff
@@ -611,6 +660,180 @@ var _ = ginkgo.Describe("Tests `appgw.ConfigBuilder`", func() {
+							HTTP: &v1beta1.HTTPIngressRuleValue{
+								Paths: []v1beta1.HTTPIngressPath{
+									{
+										Path: "/A/",
+										Backend: v1beta1.IngressBackend{
+											ServiceName: serviceNameA,
+											ServicePort: intstr.IntOrString{
+												Type:   intstr.Int,
+												IntVal: 80,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				TLS: []v1beta1.IngressTLS{
+					{
+						Hosts: []string{
+							tests.Host,
+						},
+						SecretName: tests.NameOfSecret,
+					},
+				},
+				Backend: &v1beta1.IngressBackend{
+					ServiceName: serviceNameB,
+					ServicePort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 80,
+					},
+				},
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Annotations: map[string]string{
+					annotations.IngressClassKey: annotations.ApplicationGatewayIngressClass,
+				},
+				Namespace: tests.Namespace,
+				Name:      tests.Name,
+			},
+		}
+
+		ingressWithDefaultHttpsBackend := &v1beta1.Ingress{
+			Spec: v1beta1.IngressSpec{
+				TLS: []v1beta1.IngressTLS{
+					{
+						Hosts: []string{
+							tests.Host,
+						},
+						SecretName: tests.NameOfSecret,
+					},
+				},
+				Backend: &v1beta1.IngressBackend{
+					ServiceName: serviceNameD,
+					ServicePort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 443,
+					},
+				},
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Annotations: map[string]string{
+					annotations.IngressClassKey:             annotations.ApplicationGatewayIngressClass,
+					annotations.AppGwTrustedRootCertificate: "root-cert",
+					annotations.BackendProtocolKey:          "https",
+					annotations.FirewallPolicy:              "waf-policy",
+				},
+				Namespace: tests.Namespace,
+				Name:      tests.Name,
+			},
+		}
+
+		ingressWithDefaultBackendNoMatchedRule := &v1beta1.Ingress{
+			Spec: v1beta1.IngressSpec{
+				Rules: []v1beta1.IngressRule{
+					{
+						IngressRuleValue: v1beta1.IngressRuleValue{
+							HTTP: &v1beta1.HTTPIngressRuleValue{
+								Paths: []v1beta1.HTTPIngressPath{
+									{
+										Path: "/Z/",
+										Backend: v1beta1.IngressBackend{
+											ServiceName: serviceNameZ,
+											ServicePort: intstr.IntOrString{
+												Type:   intstr.Int,
+												IntVal: 80,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				Backend: &v1beta1.IngressBackend{
+					ServiceName: serviceNameB,
+					ServicePort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 80,
+					},
+				},
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Annotations: map[string]string{
+					annotations.IngressClassKey: annotations.ApplicationGatewayIngressClass,
+				},
+				Namespace: tests.Namespace,
+				Name:      tests.Name,
+			},
+		}
+
+		// Make sure ingress default backend is used as default
+		// The no matched rule backend in this case will still be used in urlPathMaps
```

Comment:
what is the meaning of **no matched rule** ?

Reply:
> what is the meaning of **no matched rule** ?
it means no backend and service pair can be found, in this case, the rule path if defined will still be added to the pathRules

Reply:
Updated the comment into "No paired backend in this case will still be in urlPathMaps"